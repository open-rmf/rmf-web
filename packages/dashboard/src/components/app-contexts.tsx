import React from 'react';
import ResourceManager from '../managers/resource-manager';
import { defaultSettings, Settings } from '../settings';

/* Declares the ResourcesContext which contains the resources used on the app*/
export const ResourcesContext = React.createContext<ResourceManager | undefined>(undefined);

export const SettingsContext = React.createContext(defaultSettings());

export interface AppController {
  setSettings: React.Dispatch<React.SetStateAction<Settings>>;
  saveSettings(settings: Settings): void;
  showSettings(show: boolean): void;
  toggleSettings(): void;
  showHelp(show: boolean): void;
  toggleHelp(): void;
  showHotkeysDialog(show: boolean): void;
  toggleHotkeysDialog(): void;
  /**
   * FIXME: Move this to settings, this sets if we should show the tooltip icon, and not trigger
   * a tooltip to show.
   */
  showTooltips(show: boolean): void;
  /**
   * FIXME: Move this to settings, this sets if we should show the tooltip icon, and not trigger
   * a tooltip to show.
   */
  toggleTooltips(): void;
}

export interface Tooltips {
  showTooltips: boolean;
}

export const TooltipsContext = React.createContext<Tooltips>({
  showTooltips: true,
});

export const AppControllerContext = React.createContext<AppController>({
  setSettings: () => {},
  saveSettings: () => {},
  showSettings: () => {},
  toggleSettings: () => {},
  showHelp: () => {},
  toggleHelp: () => {},
  showHotkeysDialog: () => {},
  toggleHotkeysDialog: () => {},
  showTooltips: () => {},
  toggleTooltips: () => {},
});

export interface AppContent {
  tabNames: string[];
}
export const AppContentContext = React.createContext<AppContent>({ tabNames: ['Building'] });
