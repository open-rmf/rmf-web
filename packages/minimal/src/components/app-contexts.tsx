import React from 'react';
import appConfig, { AppConfig } from '../app-config';
import dataConfig, { DataConfig } from '../config/data-config';
import { ResourceManager } from 'rmf-tools';

/* Declares the ResourcesContext which contains the resources used on the app*/
export const ResourcesContext = React.createContext<ResourceManager | undefined>(undefined);

export const AppConfigContext = React.createContext<AppConfig>(appConfig);

export const DataConfigContext = React.createContext<DataConfig>(dataConfig);
