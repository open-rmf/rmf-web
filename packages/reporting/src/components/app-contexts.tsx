import React from 'react';
import appConfig, { AppConfig } from '../app-config';

export const AppConfigContext = React.createContext<AppConfig>(appConfig);
