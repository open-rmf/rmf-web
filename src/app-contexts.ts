import { createContext } from 'react';
import { ResourceConfigurationsType } from './resource-manager';

/* Declares the ResourcesContext which contains the resources used on the app*/
export const ResourcesContext = createContext<ResourceConfigurationsType>({});
