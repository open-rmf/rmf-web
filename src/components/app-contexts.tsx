import React from 'react';
import { ResourceConfigurationsType } from '../resource-manager';
import { defaultSettings, Settings } from '../settings';
import { NotificationBarContext, NotificationBarProps } from './notification-bar';

/* Declares the ResourcesContext which contains the resources used on the app*/
export const ResourcesContext = React.createContext<ResourceConfigurationsType>({});

export const SettingsContext = React.createContext(defaultSettings());

export interface AppContextProviderProps extends React.PropsWithChildren<{}> {
  settings: Settings;
  notificationDispatch: React.Dispatch<React.SetStateAction<NotificationBarProps | null>>;
}

export function AppContextProvider(props: AppContextProviderProps): React.ReactElement {
  const { settings, notificationDispatch, children } = props;
  return (
    <SettingsContext.Provider value={settings}>
      <NotificationBarContext.Provider value={notificationDispatch}>
        {children}
      </NotificationBarContext.Provider>
    </SettingsContext.Provider>
  );
}
