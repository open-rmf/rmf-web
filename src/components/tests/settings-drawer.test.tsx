import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import {
  loadSettings,
  TrajectoryAnimation,
  Settings,
  TrajectoryDiameter,
  TrajectoryColor,
} from '../../settings';
import { SettingsContext } from '../app-contexts';
import SettingsDrawer from '../drawers/settings-drawer';

const mount = createMount();

describe('Settings Drawer', () => {
  let settings = loadSettings();
  let onSettingsChange: jest.Mock;

  beforeEach(() => {
    onSettingsChange = jest.fn();
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <SettingsContext.Provider value={settings}>
        <SettingsDrawer settings={settings} onSettingsChange={onSettingsChange} open={true} />
      </SettingsContext.Provider>,
    );
    expect(toJson(wrapper.dive())).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should call onSettingsChange function when handleTrajectoryAnimationChange is called', () => {
    const newSettings: Settings = {
      ...settings,
      trajectoryAnimation: TrajectoryAnimation.Outline,
    };
    const mockEvent = { target: { value: TrajectoryAnimation.Outline } };
    const wrapper = mount(
      <SettingsContext.Provider value={newSettings}>
        <SettingsDrawer
          settings={settings}
          onSettingsChange={(newSettings) => onSettingsChange(newSettings)}
          open={true}
        />
      </SettingsContext.Provider>,
    );

    // Accessing the 'outline' animation @ the fourth input field
    wrapper.find('input').find({ name: 'Outline' }).simulate('change', mockEvent);
    expect(onSettingsChange).toHaveBeenCalledTimes(1);
  });

  it('should call onSettingsChange function when handleTrajectoryDiameterChange is called', () => {
    const newSettings: Settings = {
      ...settings,
      trajectoryDiameter: TrajectoryDiameter.FixSize,
    };
    const mockEvent = { target: { value: TrajectoryDiameter.FixSize } };
    const wrapper = mount(
      <SettingsContext.Provider value={newSettings}>
        <SettingsDrawer
          settings={settings}
          onSettingsChange={(newSettings) => onSettingsChange(newSettings)}
          open={true}
        />
      </SettingsContext.Provider>,
    );

    // Accessing the 'FixSize' animation @ the fifth input field
    wrapper.find('input').find({ name: 'Fix Size' }).simulate('change', mockEvent);
    expect(onSettingsChange).toHaveBeenCalledTimes(1);
  });

  it('should call onSettingsChange function when handleTrajectoryColorChange is called', () => {
    const newSettings: Settings = {
      ...settings,
      trajectoryColor: TrajectoryColor.Theme,
    };
    const mockEvent = { target: { value: TrajectoryColor.Theme } };
    const wrapper = mount(
      <SettingsContext.Provider value={newSettings}>
        <SettingsDrawer
          settings={settings}
          onSettingsChange={(newSettings) => onSettingsChange(newSettings)}
          open={true}
        />
      </SettingsContext.Provider>,
    );

    // Accessing the 'Theme' animation @ the seventh input field
    wrapper.find('input').find({ name: 'Theme' }).simulate('change', mockEvent);
    expect(onSettingsChange).toHaveBeenCalledTimes(1);
  });
});
