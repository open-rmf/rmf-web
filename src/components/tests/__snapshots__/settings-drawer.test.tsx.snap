// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Drawer should render correctly 1`] = `
<WithStyles(ForwardRef(Drawer))
  PaperProps={
    Object {
      "className": "makeStyles-drawer-1",
    }
  }
  anchor="right"
  open={true}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-heading-5"
      item={true}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h6"
      >
        Settings
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(IconButton))
        className="makeStyles-button-6"
        id="closeDrawerButton"
        onClick={[Function]}
      >
        <CloseIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(FormControl))
    component="fieldset"
  >
    <WithStyles(ForwardRef(FormLabel))
      className="makeStyles-legendLabel-2"
      component="legend"
    >
      Trajectory Animation
    </WithStyles(ForwardRef(FormLabel))>
    <WithStyles(ForwardRef(Divider)) />
    <ForwardRef(RadioGroup)
      className="makeStyles-trajGroup-3"
      onChange={[Function]}
      value={2}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        className="makeStyles-flexBasis-4"
        control={<ForwardRef(WithStyles) />}
        key="0"
        label="None"
        name="None"
        value={0}
      />
      <WithStyles(ForwardRef(FormControlLabel))
        className="makeStyles-flexBasis-4"
        control={<ForwardRef(WithStyles) />}
        key="1"
        label="Fill"
        name="Fill"
        value={1}
      />
      <WithStyles(ForwardRef(FormControlLabel))
        className="makeStyles-flexBasis-4"
        control={<ForwardRef(WithStyles) />}
        key="2"
        label="Follow"
        name="Follow"
        value={2}
      />
      <WithStyles(ForwardRef(FormControlLabel))
        className="makeStyles-flexBasis-4"
        control={<ForwardRef(WithStyles) />}
        key="3"
        label="Outline"
        name="Outline"
        value={3}
      />
    </ForwardRef(RadioGroup)>
    <WithStyles(ForwardRef(FormLabel))
      className="makeStyles-legendLabel-2"
      component="legend"
    >
      Trajectory Diameter
    </WithStyles(ForwardRef(FormLabel))>
    <WithStyles(ForwardRef(Divider)) />
    <ForwardRef(RadioGroup)
      className="makeStyles-trajGroup-3"
      onChange={[Function]}
      value={1}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        className="makeStyles-flexBasis-4"
        control={<ForwardRef(WithStyles) />}
        key="0"
        label="Fix Size"
        name="Fix Size"
        value={0}
      />
      <WithStyles(ForwardRef(FormControlLabel))
        className="makeStyles-flexBasis-4"
        control={<ForwardRef(WithStyles) />}
        key="1"
        label="Robot Size"
        name="Robot Size"
        value={1}
      />
    </ForwardRef(RadioGroup)>
    <WithStyles(ForwardRef(FormLabel))
      className="makeStyles-legendLabel-2"
      component="legend"
    >
      Trajectory Color
    </WithStyles(ForwardRef(FormLabel))>
    <WithStyles(ForwardRef(Divider)) />
    <ForwardRef(RadioGroup)
      className="makeStyles-trajGroup-3"
      onChange={[Function]}
      value={1}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        className="makeStyles-flexBasis-4"
        control={<ForwardRef(WithStyles) />}
        key="0"
        label="Theme"
        name="Theme"
        value={0}
      />
      <WithStyles(ForwardRef(FormControlLabel))
        className="makeStyles-flexBasis-4"
        control={<ForwardRef(WithStyles) />}
        key="1"
        label="Robot Color"
        name="Robot Color"
        value={1}
      />
      <WithStyles(ForwardRef(FormControlLabel))
        className="makeStyles-flexBasis-4"
        control={<ForwardRef(WithStyles) />}
        key="2"
        label="Shades"
        name="Shades"
        value={2}
      />
    </ForwardRef(RadioGroup)>
  </WithStyles(ForwardRef(FormControl))>
</WithStyles(ForwardRef(Drawer))>
`;
